### PR TITLE
chore(ci): Bump actions/labeler from 4 to 5

### PR DESCRIPTION
**Pull request by yin1999**
#### Summary

Bump actions/labeler from 4 to 5.

The new [input: `sync-label`](https://github.com/actions/labeler?tab=readme-ov-file#inputs) is added to automatically remove unmatched labels (which is configured in label matching entries)

#### Test results and supporting details

We can test the matching rule with the following script:

```js
import { minimatch } from "minimatch";

const filePath = ".github/labeler.yaml";
const pattern = ".github/**";

console.log(minimatch(filePath, pattern));
```

#### Related issues and pull requests

- mdn/content: mdn/content#32644
- mdn/translated-content: mdn/translated-content#18402

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -1,46 +1,83 @@
...
+      - any-glob-to-any-file:
+          - "package-lock.json"
 docs :writing_hand::
-  - "**/*.md"
+  - changed-files:
+      - any-glob-to-any-file:
+          - "**/*.md"
 infra :building_construction::
-  - ".*"
-  - "/*.js"
```
Comment (yin1999) [inline .github/labeler.yml:31]: replaced with `**/*.js` to match all `js` files. Currently, there is no js files in this repo.

File: .github/labeler.yml [hunk 2]
```diff
@@ -1,46 +1,83 @@
...
-  - "*.ts"
```
Comment (yin1999) [inline .github/labeler.yml:32]: replaced with `**/*.ts`, as `*.ts` can only math the ts files under the root dir.

File: .github/labeler.yml [hunk 3]
```diff
@@ -1,46 +1,83 @@
...
-  - "**/*.d.ts"
-  - "/index.js"
```
Comment (yin1999) [inline .github/labeler.yml:34]: removed, as the `*.js` can match it

**Review by bsmth — APPROVED** [review]
Thank you, @yin1999!

I've reviewed the other two related PRs, so I'm leaving a +1 on this, also. For other reviewers: the diff is mostly a structural change in the config files for the bumped labeler and LGTM 👍🏻

**Review by queengooborg — APPROVED** [review]
Thanks for tackling this migration!

**Pull request merged**